src/main/java/org/junit/runner/notification/RunListener.java

```diff
@@ -122,6 +123,15 @@ public void testAssumptionFailure(Failure failure) {
     public void testIgnored(Description description) throws Exception {
     }
 
+    /**
+     * Called when some services loaded using {@link org.junit.internal.ServiceLoaderWrapper}
+     * @param serviceType a type of the loaded services
+     * @param services loaded services
+     *
+     * @since 4.12
+     */
+    public <T> void servicesLoaded(Class<T> serviceType, List<T> services) {
+    }
```

Comment:
@marcphilipp I believe David or I asked for this method, so that a RunListener could print a message indicating that a service has loaded. Otherwise, you get "magic" new behavior with no notification.